isisd/isis_lsp.c

```diff
@@ -2297,21 +2297,39 @@ static int lsp_handle_adj_state_change(struct isis_adjacency *adj)
 /*
  * Iterate over all IP reachability TLVs in a LSP (all fragments) of the given
  * address-family and MT-ID.
+ *
+ * algorithm parameter: was added to support flex-algo. For non-flex-algo
+ * usage, algorithm should be 0. For flex-algo usage, algorithm should be
+ * in the range 128-255.
  */
```

Comment:
This is a wrong modification of isis_lsp_iterate_ip_reach(). In fact, these functions was originally defined to ease loop other all LSPs. This is generic functions that could be used in various place and which execute a callback function pass as arguments. It is use in various place (isis_sr.c, isis_te.c at least). For flex algo, instead of modify the generic function, please call the function with appropriate callback function to process Flex algo. The callback function could return LSP_ITER_CONTINUE or LSP_ITER_STOP to control how the iteration on LSP is done.